### PR TITLE
Add Urban Terror

### PR DESCRIPTION
**Pull request by Juli199696**
# Description

Added a Gameserver Egg for the Game Urban Terror (4.3.4)
## Checklist for all submissions

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you tested and reviewed your changes with confidence that everything works?
* [x] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

* [x] You verify that the start command applied does not use a shell script
  * [ ] If some script is needed then it is part of a current yolk or a PR to add one
* [x] The egg was exported from the panel

## New egg Submissions

1. [x] Does your submission pass tests (server is connectable)?
2. [ ] Does your egg use a custom docker image?
    * [ ] Have you tried to use a generic image?
    * [ ] Did you PR the necessary changes to make it work?
3. [x] Have you added the egg to the main README.md and any other README files in subdirectories of the egg (e.g /game_eggs) according to the alphabetical order?
4. [x] Have you added a unique README.md for the egg you are adding according to the alphabetical order?
5. [x] You verify that the start command applied does not use a shell script
    * [ ] If some script is needed then it is part of a current yolk or a PR to add one
6. [x] The egg was exported from the panel

### CODE REVIEW

**Comment by Juli199696** [comment]
This is my first real PR on Github, so please have mercy and explain me what i can do better :)

File: game_eggs/urbanterror/egg-urban-terror4-3-4.json [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        "version": "PTDL_v2",
+        "update_url": null
+    },
+    "exported_at": "2024-02-09T19:34:16+00:00",
+    "name": "Urban Terror 4.3.4",
+    "author": "Juli199696@web.de",
+    "description": "Urban Terror started as a realism based \"total conversion\" mod for Quake III Arena. Nowadays, Urban Terror is played using the free, stand-alone, ioquake3 engine modified, which doesn't require Quake III Arena, in addition to being an add-on to id Software's Quake III Arena.",
+    "features": null,
+    "docker_images": {
+        "quay.io\/pterodactyl\/core:glibc": "quay.io\/pterodactyl\/core:glibc"
```
Comment (QuintenQVD0) [inline game_eggs/urbanterror/egg-urban-terror4-3-4.json:13]: This is dead old and should not be used anymore. Please use the debian image.

File: game_eggs/urbanterror/egg-urban-terror4-3-4.json [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        "startup": "{\r\n    \"done\": \"------- Game Initialization -------\",\r\n    \"userInteraction\": []\r\n}",
+        "logs": "{\r\n    \"custom\": false,\r\n    \"location\": \"\/home\/container\/q3ut4\/games.log\"\r\n}",
```
Comment (QuintenQVD0) [inline game_eggs/urbanterror/egg-urban-terror4-3-4.json:20]: This should not be used anymore. Just put {}

File: game_eggs/urbanterror/egg-urban-terror4-3-4.json [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        "startup": "{\r\n    \"done\": \"------- Game Initialization -------\",\r\n    \"userInteraction\": []\r\n}",
```
Comment (QuintenQVD0) [inline game_eggs/urbanterror/egg-urban-terror4-3-4.json:19]: Userinteraction can be removed 0.7 leftover. Does not work anymore

File: game_eggs/urbanterror/egg-urban-terror4-3-4.json [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+            "container": "debian:buster-slim",
```
Comment (QuintenQVD0) [inline game_eggs/urbanterror/egg-urban-terror4-3-4.json:26]: Please use a parker offical yols debian installer here

File: game_eggs/urbanterror/egg-urban-terror4-3-4.json [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "required|integer|max:32",
+            "field_type": "text"
+        },
+        {
+            "name": "RCON Password",
+            "description": "Set the Rcon Password for Admin commands",
+            "env_variable": "RCON_PASSWORD",
+            "default_value": "PleaseChangeMe!",
```
Comment (QuintenQVD0) [inline game_eggs/urbanterror/egg-urban-terror4-3-4.json:85]: Please leave the default empry for security reasons

File: game_eggs/urbanterror/egg-urban-terror4-3-4.json [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+            "field_type": "text"
+        },
+        {
+            "name": "Server Password",
+            "description": "If you want to set a password to connect on the server set this here.",
+            "env_variable": "SERVER_PASSWORD",
+            "default_value": "",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "max:20",
```
Comment (QuintenQVD0) [inline game_eggs/urbanterror/egg-urban-terror4-3-4.json:98]: This should also have string

File: game_eggs/urbanterror/egg-urban-terror4-3-4.json [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+            "field_type": "text"
+        },
+        {
+            "name": "Gametype \/ Gamemode",
+            "description": "Sets the Gamemode you want to play. \r\n0 = Free For All, \r\n1 = Last Man Standing, \r\n3 = Team DeathMatch, \r\n4 = Team Survivor, \r\n5 = Follow The Leader, \r\n6 = Capture And Hold, \r\n7 = Capture The Flag, \r\n8 = Bomb Mode, \r\n9 = Jump, \r\n10 = Freeze Tag, \r\n11 = Gun Game",
+            "env_variable": "SERVER_GAMETYPE",
+            "default_value": "3",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "required|integer|max:11",
```
Comment (QuintenQVD0) [inline game_eggs/urbanterror/egg-urban-terror4-3-4.json:118]: There only x amount of options so use in: here

File: game_eggs/urbanterror/README.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+
+## Game Infos
+Gamemodes: 0 = Free For All, 1 = Last Man Standing, 3 = Team DeathMatch, 4 = Team Survivor, 5 = Follow The Leader, 6 = Capture And Hold, 7 = Capture The Flag, 8 = Bomb Mode, 9 = Jump, 10 = Freeze Tag, 11 = Gun Game
+You can edit the Server.cfg in /home/container/q3ut4/server.cfg as you wish after creating the server. 
+
+## Known Errors
+Updater will crash when not having enough RAM or SWAP available (at least 1024MiB RAM or 512MiB RAM with 512MiB SWAP)
+
+## Docker Infos
```
Comment (QuintenQVD0) [inline game_eggs/urbanterror/README.md:33]: All this and bellow is  nog needed (all images)

**Review by QuintenQVD0 — CHANGES_REQUESTED** [review]

File: game_eggs/urbanterror/egg-urban-terror4-3-4.json (hunk 1 above)
Comment (Juli199696) [reply game_eggs/urbanterror/egg-urban-terror4-3-4.json:13]: Using current Debian Yolk now from ghcr.io/parkervcp/installers:debian

File: game_eggs/urbanterror/egg-urban-terror4-3-4.json (hunk 2 above)
Comment (Juli199696) [reply game_eggs/urbanterror/egg-urban-terror4-3-4.json]: Deleted unused Logging.

File: game_eggs/urbanterror/egg-urban-terror4-3-4.json (hunk 3 above)
Comment (Juli199696) [reply game_eggs/urbanterror/egg-urban-terror4-3-4.json]: Removed Useraction.

**Comment by Juli199696** [comment]
Updated all requested parts to fit the current schemata and newest debian yolk. Updated Readme.md to fit the others.

File: game_eggs/urbanterror/egg-urban-terror4-3-4.json (hunk 1 above)
Comment (Juli199696) [reply game_eggs/urbanterror/egg-urban-terror4-3-4.json]: I changed it now to ghcr.io/parkervcp/yolks:debian

I remember replacing it yesterday but it seems i forgot to click in save after that and reinstalled the server. It works now with the debian yolk. Not with the installer debian :)

**Comment by QuintenQVD0** [comment]
I did not forget you. Just a litte bussy rn.

**Comment by Juli199696** [comment]
Yeah no pressure tho. I just though it would be the "normal" way to re request the review. No hurry mate 👍

**Comment by Juli199696** [comment]
> * Rename to just Urban terror as that is for I can see the current only existing version
> * Move to an installer image 
Yeah the gamename should not change anymore soon as they are working on a remake of the whole game anyway. It did tho can happen that they push a security update if needed. I set up the script to skip (accept the game updater eula and select the version of game) so it always downloads the latest version of the game. 
 
I just used "container": "debian:bullseye-slim" because the Updater for some reason did not work with the installer image.

**Comment by QuintenQVD0** [comment]
> > * Rename to just Urban terror as that is for I can see the current only existing version
> > * Move to an installer image
> 
> Yeah the gamename should not change anymore soon as they are working on a remake of the whole game anyway. It did tho can happen that they push a security update if needed. I set up the script to skip (accept the game updater eula and select the version of game) so it always downloads the latest version of the game. I just used "container": "debian:bullseye-slim" because the Updater for some reason did not work with the installer image.

there the same image as the installer on is based on that one? just with like 8 extra packages?

**Comment by Juli199696** [comment]
I just tested your changes and it works perfectly tho. So something was wrong/off on my end. Sorry for the confusion. 
Btw, you can test if the server is actually running by adding bots (while setup) and see that they play in the gameserver console as the log is logging ingame kills by default.

**Comment by QuintenQVD0** [comment]
Something seems to fail with the i386 version download (I did select the normal x64 version)
https://pteropaste.com/2kul0mv6kp5j

**Comment by Juli199696** [comment]
It looks like they have removed the i386 version from the Updater. In the Full Zip of the game both files are existing. So you or me can just remove the selection for the i386 file and it will work fine.

**Comment by QuintenQVD0** [comment]
> It looks like they have removed the i386 version from the Updater as 32bit processors are depricated anyway. In the Full Zip of the game both files are existing. So you or me can just remove the selection for the i386 file and it will work fine.

Does the panel also think for you that the game has crashed if you do in the console `cmdlist`?

**Comment by Juli199696** [comment]
I never tried that command but yes it does. The Server still runs tho. Seems like a bug in Pyterodactyl? Maybe he detect one of the commands as "Crash or Quit" command and thinks it stopped?

**Review by QuintenQVD0 — APPROVED** [review]
LGTM! Thanks

**Pull request merged**